component.js

```diff
@@ -200,33 +206,35 @@ function factory (options) {
       debug = _shouldComponentUpdate.debug(pattern, logFn);
     }
   }
-
-  function createDefaultArguments (displayName, mixins, render) {
-
-    // (render)
-    if (typeof displayName === 'function') {
-      render      = displayName;
-      mixins      = [];
-      displayName = void 0;
+  
+  var NO_RENDER = "Component definition requires `render` function as its argument"
+
+  // [displayName], [mixins = []], render, [decorate]
+  function createDefaultArguments (/* args */) {
+    var args = Array.prototype.slice.call(arguments);
+    var currArg = 0, argCount = args.length;;
+    
+    var displayName, mixins = [], render, decorate;
+    
+    if (currArg == argCount) throw NO_RENDER;
+    if (typeof args[currArg] === 'string' || args[currArg] instanceof String) {
```

Comment:
Yep, `new String('...')` are valid JS strings, though their `typeof` is `object`. It's more likely that you know the answer if the question was if we should support such strings.